Source/Scene/Model.js

```diff
@@ -3808,19 +3967,42 @@ define([
             var boundingVolume;
 
             if (passes.render) {
+
+                // The actual render commands
                 for (i = 0; i < length; ++i) {
                     nc = nodeCommands[i];
                     if (nc.show) {
-                        var command = nc.command;
-                        commandList.push(command);
+                        commandList.push(nc.command);
 
-                        boundingVolume = command.boundingVolume;
+                        boundingVolume = nc.command.boundingVolume;
                         if (frameState.mode === SceneMode.SCENE2D &&
                             (boundingVolume.center.y + boundingVolume.radius > idl2D || boundingVolume.center.y - boundingVolume.radius < idl2D)) {
                             commandList.push(nc.command2D);
                         }
                     }
                 }
+
+                if (this.highlight) {
+
+                    // Only render the highlight commands if we have sufficient stencil bits.
+                    if (context.stencilBits > 0) {
+                        // highlight commands second.
+                        for (i = 0; i < length; ++i) {
+                            nc = nodeCommands[i];
+                            if (nc.show) {
+                                commandList.push(nc.highlightCommand);
+                                boundingVolume = nc.command.boundingVolume;
+                                if (frameState.mode === SceneMode.SCENE2D &&
+                                    (boundingVolume.center.y + boundingVolume.radius > idl2D || boundingVolume.center.y - boundingVolume.radius < idl2D)) {
+                                    commandList.push(nc.highlightCommand2D);
+                                }
+                            }
+                        }
+                    }
+                    else {
+                        console.log("Model highlighting not supported, stencilBits = " + context.stencilBits + ".  Request a stencil buffer when initializing the Viewer");
```

Comment:
Instead of this `console.log`, let's just throw an exception when highlight is enabled and this is not supported; mirror how polygons on terrain work.